deployer/src/handlers/mod.rs

```diff
@@ -0,0 +1,42 @@
+use axum::{handler::Handler, headers::HeaderMapExt, routing::post, Router};
+use http::Uri;
+use shuttle_common::{
+    backends::{
+        auth::{AuthPublicKey, JwtAuthenticationLayer, ScopedLayer},
+        headers::XShuttleAccountName,
+        metrics::TraceLayer,
+    },
+    claims::Scope,
+    request_span,
+};
+
+use tracing::field;
+
+mod deployment;
+mod error;
+
+pub async fn make_router(auth_uri: Uri) -> Router {
+    Router::new()
+        .route(
+            "/deploy/:project_name",
+            post(deployment::deploy_project.layer(ScopedLayer::new(vec![Scope::DeploymentPush]))),
+        )
+        .layer(JwtAuthenticationLayer::new(AuthPublicKey::new(auth_uri)))
+        // This route should be below the auth bearer since it does not need authentication
+        .layer(
+            TraceLayer::new(|request| {
+                let account_name = request
+                    .headers()
+                    .typed_get::<XShuttleAccountName>()
```

Comment:
We might also no longer have this :smile:

Reply:
Indeed. Removed it.